qemu/tests/qemu_guest_agent.py

```diff
@@ -1827,6 +1827,190 @@ def ip_addr_check(session, mac_addr, ret_list, if_index, if_name):
+                    desprefixlen = 0  # default desprefixlen should be 0
+                    processed_routes.append(
+                        {
+                            "iface": iface,
+                            "destination": destination,
+                            "desprefixlen": desprefixlen,
+                            "nexthop": nexthop,
+                            "metric": metric,
+                        }
+                    )
+                    continue
+                match = route_pattern.match(line)
+                if match:
+                    destination = match.group("destination")
+                    iface = match.group("iface")
+                    metric = match.group("metric")
+
+                    if "/" in destination:
+                        destination = destination.split("/")[0]
+                        desprefixlen = 64
+                    else:
+                        desprefixlen = 128
+                    nexthop = "::"
+                    processed_routes.append(
+                        {
+                            "iface": iface,
+                            "destination": destination,
+                            "desprefixlen": desprefixlen,
+                            "nexthop": nexthop,
+                            "metric": metric,
+                        }
+                    )
+            return processed_routes
+
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Getting route info via guest-agent API.", LOG_JOB.info)
+        self.gagent.get_network_route()
+        qga_route_info = self.gagent.get_network_route()
+
+        error_context.context("Getting route info from inside the guest.", LOG_JOB.info)
+        ipv4_routes = parse_ipv4_route(session.cmd_output(self.params["cmd_ipv4route"]))
+        ipv6_routes = parse_ipv6_route(session.cmd_output(self.params["cmd_ipv6route"]))
+        guest_route_info = ipv4_routes + ipv6_routes
+
+        error_context.context(
+            "Compare the route info from qga and guest.", LOG_JOB.info
+        )
+        ipv4_keys = list(self.params["ipv4_keys"])
+        ipv6_keys = list(self.params["ipv6_keys"])
+
+        for guest_entry in guest_route_info:
+            matched_qga_entry = None
+            for qga_entry in qga_route_info:
+                if "mask" in guest_entry:
+                    if all(
+                        guest_entry.get(key) == qga_entry.get(key)
+                        for key in ["iface", "destination"]
+                    ):
+                        matched_qga_entry = qga_entry
+                        break
+                else:
+                    if all(
+                        guest_entry.get(key) == qga_entry.get(key)
+                        for key in ["iface", "destination"]
+                    ):
+                        matched_qga_entry = qga_entry
+                        break
```

Comment:
The code logic in if and else is exactly the same, can we merge them?

Reply:
it's funny, I'll test it later.

Reply:
Hi @6-dehan Any update about this part?

Reply:
(1/1) Host_RHEL.m9.u6.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.6.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_get_network_route.q35: PASS (68.10 s)

extra code. delete.